feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/presentation/proxy/sign/RealProxySigningPresenter.kt

```diff
@@ -0,0 +1,105 @@
+package io.novafoundation.nova.feature_account_impl.presentation.proxy.sign
+
+import android.text.SpannableStringBuilder
+import io.novafoundation.nova.common.data.storage.Preferences
+import io.novafoundation.nova.common.resources.ContextManager
+import io.novafoundation.nova.common.resources.ResourceManager
+import io.novafoundation.nova.common.utils.colorSpan
+import io.novafoundation.nova.common.utils.formatting.spannable.SpannableFormatter
+import io.novafoundation.nova.common.utils.toSpannable
+import io.novafoundation.nova.feature_account_api.domain.model.MetaAccount
+import io.novafoundation.nova.feature_account_api.domain.model.ProxyAccount
+import io.novafoundation.nova.feature_account_api.presenatation.account.proxy.ProxySigningPresenter
+import io.novafoundation.nova.feature_account_impl.R
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+private const val KEY_DONT_SHOW_AGAIN = "proxy_sign_warning_dont_show_again"
+
+class RealProxySigningPresenter(
+    private val contextManager: ContextManager,
+    private val resourceManager: ResourceManager,
+    private val preferences: Preferences
+) : ProxySigningPresenter {
+
+    override suspend fun requestResume(proxiedMetaAccount: MetaAccount, proxyMetaAccount: MetaAccount): Boolean = withContext(Dispatchers.Main) {
+        if (notNeedToShowWarning(proxiedMetaAccount)) {
+            return@withContext true
+        }
+
+        val resumingAllowed = suspendCoroutine { continuation ->
+            ProxySignWarningBottomSheet(
+                context = contextManager.getActivity()!!,
+                subtitle = formatSubtitleForWarning(proxyMetaAccount),
+                onFinish = {
+                    continuation.resume(it)
+                },
+                dontShowAgain = { dontShowAgain(proxiedMetaAccount) }
+            ).show()
+        }
+
+        return@withContext resumingAllowed
```

Comment:
No `return` needed here

Reply:
No we need return here since we can click continue or cancel.

For cancel case we will stop signing
<img width="564" alt="image" src="https://github.com/novasamatech/nova-wallet-android/assets/107959809/360356d1-aef9-4b5a-acc5-aadeac13361a">